### PR TITLE
NFC: fixes & improvements

### PR DESCRIPTION
**Pull request by yobabyte**
# What's new

- Fixed reading for Mifare Ultralight tags - checks reading boundary and discards excess.
- Fixed FAST_READ command emulation for Mifare Ultralight - handles requested page range correctly.
- Improved Mifare Ultralight emulation stability - increased log level for some messages, because frequent calls to furi_hal_console_put() somehow directly affects emulation stability (delays due to locks or something, dunno).
- Implemented non-strict PWD_AUTH command emulation for NTAG21x - performs authentication if PACK is not set to default value, otherwise accepts any passwords.
- _Plastic figures and cards with NTAG215 from a well-known Japanese company are now correctly read and emulated_ :smirk:

# Verification 

- Build and download.
- Read some NTAG21n tag:
  - Save, check number of pages in saved file (e.g. NTAG215 should contain pages from 0 to 134 inclusive)
  - Emulate that tag and try FAST_READ command (e.g. 3A 00 00 should return first 4 bytes of UID)
  - Replace PWD and PACK with actual values in saved file, emulate and try PWD_AUTH command:
  ```
  # NTAG215 primer
  Page 133: 00 E6 AC 0C
  Page 134: DE AD 00 00
  
  # Command: 1B 00 E6 AC 0C
  # Response: DE AD (PACK)
  
  # Command: 1B F1 F2 F3 F4
  # Response: 00 (NAK)
  ```

# Checklist (do not modify)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: applications/nfc/nfc_worker.c [hunk 1]
```diff
@@ -636,7 +636,7 @@ void nfc_worker_emulate_mifare_ul(NfcWorker* nfc_worker) {
                             break;
                         }
                     } else {
-                        FURI_LOG_W(TAG, "Not valid command: %02X", rx_buff[0]);
+                        FURI_LOG_D(TAG, "Not valid command: %02X", rx_buff[0]);
```
Comment (DrZlo13) [inline applications/nfc/nfc_worker.c]: I propose to completely remove logging from this function and leave a comment about the side effects of calling a blocking function.

File: applications/nfc/nfc_worker.c [hunk 2]
```diff
@@ -653,7 +653,7 @@ void nfc_worker_emulate_mifare_ul(NfcWorker* nfc_worker) {
                 nfc_worker->callback(nfc_worker->context);
             }
         }
-        FURI_LOG_W(TAG, "Can't find reader");
+        FURI_LOG_D(TAG, "Can't find reader");
```
Comment (DrZlo13) [inline applications/nfc/nfc_worker.c]: Same here

**Review by DrZlo13 — DISMISSED** [review]

File: applications/nfc/nfc_worker.c (hunk 1 above)
Comment (yobabyte) [reply applications/nfc/nfc_worker.c]: Done.

File: applications/nfc/nfc_worker.c (hunk 2 above)
Comment (yobabyte) [reply applications/nfc/nfc_worker.c]: Done.

File: applications/nfc/nfc_worker.c [hunk 3]
```diff
@@ -613,14 +613,16 @@ void nfc_worker_emulate_mifare_ul(NfcWorker* nfc_worker) {
...
+        // DO NOT call any blocking functions (e.g. FURI_LOG_*) in this loop,
+        // as any delay will negatively affect the stability of the emulation.
         if(furi_hal_nfc_listen(
                data->nfc_data.uid,
                data->nfc_data.uid_len,
                data->nfc_data.atqa,
                data->nfc_data.sak,
                true,
                200)) {
-            FURI_LOG_D(TAG, "Anticollision passed");
```
Comment (skotopes) [inline applications/nfc/nfc_worker.c:623]: I guess we can leave logging with level set to debug. @gornekich what do you think?

Comment (gornekich) [reply applications/nfc/nfc_worker.c:623]: I agree, it's better to leave logs at Debug level. @ghettorce could you make it?

**Review by skotopes — APPROVED** [review]

**Pull request merged**